src/components/Construction/index.tsx

```diff
@@ -0,0 +1,275 @@
+import { RIGHT_BAR_WIDTH } from "@/constants/designPage";
+import { Box, Center, Button } from "@chakra-ui/react";
+import { useState, useRef, useEffect } from "react";
+import Konva from "konva";
+import { useDispatch } from "react-redux";
+import { Stage, Layer, Line, Rect, Text, Image } from "react-konva";
+import { useStoreSelector } from "@/store/hooks";
+import {
+  COORDINATES_AREA_WIDTH,
+  IMAGE_COVERAGE,
+  DRAW_DELAY,
+  PIXEL_RATIO,
+} from "@/constants/construction";
+import { findDistinctColor } from "@/utils/findDistinctColor";
+import { changeConstructionPdfSrc } from "@/store/reducer/constructionSlice";
+import {
+  switchConstructionOpen,
+  switchConstructionMounted,
+} from "@/store/reducer/buttonToggleSlice";
+
+interface Coordinates {
+  x: number;
+  y: number;
+  width: number;
+  height: number;
+  text: string;
+}
+
+const Construction = () => {
+  const dispatch = useDispatch();
+  const isConstructionOpen = useStoreSelector((state) => state.buttonToggle.isConstructionOpen);
+  const pdfRef = useRef<Konva.Stage>(null);
+  const imgSrc = useStoreSelector((state) => state.construction.constructionSrc);
+  const constructionInfo = useStoreSelector((state) => state.construction.constructionInfo);
+  const tiles = useStoreSelector((state) => state.tile.present.court);
+  const tilesColor = tiles.map((item) => item.color);
+  const distinctColor = findDistinctColor(tilesColor);
+
+  const { beginPointX, beginPointY, endPointX, endPointY, tileSize } = constructionInfo;
+  const constructionRatio = window.innerHeight / (endPointY - beginPointY);
+  // calculate suitable font size for different design
+  const coordinateFontSize = Math.floor(3 * constructionRatio);
+  const constructionWidth = constructionRatio * (endPointX - beginPointX);
+  const constructionHeight = constructionRatio * (endPointY - beginPointY);
+  const canvasWidth =
+    constructionWidth * IMAGE_COVERAGE +
+    COORDINATES_AREA_WIDTH +
+    tileSize * constructionRatio * IMAGE_COVERAGE;
+  const canvasHeight =
+    constructionHeight * IMAGE_COVERAGE +
+    COORDINATES_AREA_WIDTH +
+    tileSize * constructionRatio * IMAGE_COVERAGE;
+
+  const [img, setImg] = useState(new window.Image());
+  useEffect(() => {
+    if (imgSrc) {
+      const loadImg = new window.Image();
+      loadImg.onload = () => {
+        setImg(loadImg);
+      };
+      loadImg.src = imgSrc;
+    }
```

Comment:
if imgSrc
  img.src = imgSrc


Anyway, this part confuses me.

Reply:
The main reason for dealing with it like this is because Konva needs a fully loaded image.
The process of drawing image in Konva is like following:
1. Create a HTMLImageElement with `constloadImg = new window.Image();`
2. In Konva, `<Image image={img} />`, Image will draw a canvas with the image at the moment of rendering. 
3. When we set `loadImg.src = imgSrc`, loadImg will begin to load asynchronourly. At this moment, image is still blank.
4. We use `loadImg.onload = () => {setImg(loadImg);};` to listen to the load event, set image to loadImg when it completes